[zh-cn] sync authorization.md (#46537)
kubernetes/website: content/zh-cn/docs/reference/access-authn-authz/authorization.md

- lavalamp
- deads2k
- liggitt
title: Authorization
content_type: concept
weight: 30
description: >
  Details of Kubernetes authorization mechanisms and supported authorization modes.
-->

<!-- overview -->

<!--
Kubernetes authorization takes place following
[authentication](/docs/reference/access-authn-authz/authentication/).
Usually, a client making a request must be authenticated (logged in) before its
request can be allowed; however, Kubernetes also allows anonymous requests in
some circumstances.

For an overview of how authorization fits into the wider context of API access
control, read
[Controlling Access to the Kubernetes API](/docs/concepts/security/controlling-access/).
-->
Kubernetes 鉴权在[身份认证](/zh-cn/docs/reference/access-authn-authz/authentication/)之后进行。

> 通常，发出请求的客户端必须经过身份验证（登录）才能允许其请求；

Reviewer: ```suggestion
通常，发出请求的客户端必须经过身份认证（登录），其请求才可以被允许；
```

但是，Kubernetes 在某些情况下也允许匿名请求。

> 有关鉴权如何融入 API 访问控制的更多的上下文信息，

Reviewer: ```suggestion
有关鉴权在 API 访问控制中的位置这类进一步的语境信息，
```

请阅读[控制对 Kubernetes API 的访问](/zh-cn/docs/concepts/security/controlling-access/)。

<!-- body -->

<!--
## Authorization verdicts {#determine-whether-a-request-is-allowed-or-denied}

Kubernetes authorization of API requests takes place within the API server.
The API server evaluates all of the request attributes against all policies,
potentially also consulting external services, and then allows or denies the
request.
-->
## 鉴权裁定   {#determine-whether-a-request-is-allowed-or-denied}

Kubernetes 对 API 请求的鉴权在 API 服务器内进行。
API 服务器根据所有策略评估所有请求属性，可能还会咨询外部服务，然后允许或拒绝该请求。

<!--
All parts of an API request must be allowed by some authorization
mechanism in order to proceed. In other words: access is denied by default.
-->
API 请求的所有部分都必须通过某种鉴权机制才能继续，
换句话说：默认情况下拒绝访问。

{{% note %}}
<!--
Access controls and policies that
depend on specific fields of specific kinds of objects are handled by
{{< glossary_tooltip text="admission controllers" term_id="admission-controller" >}}.

Kubernetes admission control happens after authorization has completed (and,
therefore, only when the authorization decision was to allow the request).
-->

> 依赖于特定对象种类的特定字段的{{< glossary_tooltip text="访问控制" term_id="admission-controller" >}}和策略由准入控制器处理。

Reviewer: ```suggestion
依赖于特定对象种类的特定字段的访问控制和策略由{{< glossary_tooltip text="准入控制器" term_id="admission-controller" >}}处理。
```

> Kubernetes 准入控制发生在鉴权完成后（因此，仅当鉴权决策是允许请求时）。

Reviewer: ```suggestion
Kubernetes 准入控制发生在鉴权完成之后（因此，仅当鉴权决策是允许请求时）。
```

  （仅限资源请求）。空字符串表示[核心 API 组](/zh-cn/docs/reference/using-api/#api-groups)。

<!--
### Request verbs and authorization {#determine-the-request-verb}

#### Non-resource requests {#request-verb-non-resource}

Requests to endpoints other than `/api/v1/...` or `/apis/<group>/<version>/...`
are considered _non-resource requests_, and use the lower-cased HTTP method of the request as the verb.

For example, making a `GET` request to endpoints like `/api` or `/healthz` would use **get** as the verb.
-->
### 请求动词和鉴权  {#determine-the-request-verb}

> #### 非资源请求

Reviewer: ```suggestion
#### 非资源请求   {#request-verb-non-resource}
```

对于 `/api/v1/...` 或 `/apis/<group>/<version>/...`
之外的端点的请求被视为**非资源请求（Non-Resource Requests）**，
并使用该请求的 HTTP 方法的小写形式作为其请求动词。

例如，对 `/api` 或 `/healthz` 这类端点的 `GET` 请求将使用 **get** 作为其动词。

<!--
#### Resource requests

To determine the request verb for a resource API endpoint, Kubernetes maps the HTTP verb
used and considers whether or not the request acts on an individual resource or a
collection of resources:
-->

> #### 资源请求

Reviewer: ```suggestion
#### 资源请求    {#resource-requests}
```

为了确定资源 API 端点的请求动词，Kubernetes 会映射所使用的 HTTP 动词，
并考虑该请求是否作用于单个资源或资源集合：

<!--
HTTP verb     | request verb
--------------|---------------
`POST`        | **create**
`GET`, `HEAD` | **get** (for individual resources), **list** (for collections, including full object content), **watch** (for watching an individual resource or collection of resources)
`PUT`         | **update**
`PATCH`       | **patch**
`DELETE`      | **delete** (for individual resources), **deletecollection** (for collections)
-->
HTTP 动词 | 请求动词
--------------|---------------
`POST`        | **create**

> `GET`, `HEAD` | **get**（for individual resources）, **list**（针对集合，包括完整的对象内容）, **watch**（用于查看单个资源或资源集合）

Reviewer: ```suggestion
`GET`, `HEAD` | **get**（针对单个资源）、**list**（针对集合，包括完整的对象内容）、**watch**（用于查看单个资源或资源集合）
```

`PUT`         | **update**
`PATCH`       | **patch**

> `DELETE`      | **delete**（针对单个资源）, **deletecollection**（针对集合）

Reviewer: ```suggestion
`DELETE`      | **delete**（针对单个资源）、**deletecollection**（针对集合）
```

* [Authorization of CertificateSigningRequests](/docs/reference/access-authn-authz/certificate-signing-requests/#authorization)
  * **approve** verb for CertificateSigningRequests, and **update** for revisions to existing approvals
 [RBAC](/docs/reference/access-authn-authz/rbac/#privilege-escalation-prevention-and-bootstrapping)
  * **bind** and **escalate** verbs on `roles` and `clusterroles` resources in the `rbac.authorization.k8s.io` API group.
-->
Kubernetes 有时使用专门的动词以对额外的权限进行鉴权。例如：

* [身份认证](/zh-cn/docs/reference/access-authn-authz/authentication/)的特殊情况
  * 对核心 API 组中 `users`、`groups` 和 `serviceaccounts` 以及 `authentication.k8s.io`
    API 组中的 `userextras` 所使用的 **impersonate** 动词。
* [RBAC](/zh-cn/docs/reference/access-authn-authz/rbac/#privilege-escalation-prevention-and-bootstrapping)
  * 对 `rbac.authorization.k8s.io` API 组中 `roles` 和 `clusterroles` 资源的 **bind**
    和 **escalate** 动词

<!--
## Authorization context

Kubernetes expects attributes that are common to REST API requests. This means
that Kubernetes authorization works with existing organization-wide or
cloud-provider-wide access control systems which may handle other APIs besides
the Kubernetes API.
-->

> ## 鉴权上下文

Reviewer: ```suggestion
## 鉴权上下文    {#authorization-context}
```

Kubernetes 需要 REST API 请求所共有的属性，
这意味着 Kubernetes 鉴权可与现有的组织范围或云提供商范围的访问控制系统配合使用，
这些系统可能处理除 Kubernetes API 之外的其他 API。

<!--
## Authorization modes  {#authorization-modules}

`AlwaysAllow`
: This mode allows all requests, which brings [security risks](#warning-always-allow). Use this authorization mode only if you do not require authorization for your API requests (for example, for testing).

`AlwaysDeny`
: This mode blocks all requests. Use this authorization mode only for testing.

`ABAC` ([attribute-based access control](/docs/reference/access-authn-authz/abac/))
: Kubernetes ABAC mode defines an access control paradigm whereby access rights are granted to users through the use of policies which combine attributes together. The policies can use any type of attributes (user attributes, resource attributes, object, environment attributes, etc).
-->

> ## 鉴权模块  {#authorization-modules}

Reviewer: ```suggestion
## 鉴权模式  {#authorization-modules}
```

`AlwaysAllow`

> : 此模式允许所有请求，但存 [安全风险](#warning-always-allow)，

Reviewer: ```suggestion
: 此模式允许所有请求，但存在[安全风险](#warning-always-allow)，
```

  仅当你的 API 请求不需要鉴权时（例如，用于测试），才使用此鉴权模式。

`AlwaysDeny`

> : 此模式阻止所有请求。仅将此鉴权模式用于测试。

Reviewer: ```suggestion
: 此模式阻止所有请求。此鉴权模式仅适用于测试。
```

<!--
`RBAC` ([role-based access control](/docs/reference/access-authn-authz/rbac/))
: Kubernetes RBAC is a method of regulating access to computer or network resources based on the roles of individual users within an enterprise. In this context, access is the ability of an individual user to perform a specific task, such as view, create, or modify a file.  
  In this mode, Kubernetes uses the `rbac.authorization.k8s.io` API group to drive authorization decisions, allowing you to dynamically configure permission policies through the Kubernetes API.

`Node`
: A special-purpose authorization mode that grants permissions to kubelets based on the pods they are scheduled to run. To learn more about the Node authorization mode, see [Node Authorization](/docs/reference/access-authn-authz/node/).

`Webhook`
: Kubernetes [webhook mode](/docs/reference/access-authn-authz/webhook/) for authorization makes a synchronous HTTP callout, blocking the request until the remote HTTP service responds to the query.You can write your own software to handle the callout, or use solutions from the ecosystem.
-->
`RBAC`（[基于角色的访问控制](/zh-cn/docs/reference/access-authn-authz/rbac/)）

> : Kubernetes RBAC 是一种根据企业内各个用户的角色来管理对计算机或网络资源的访问权限的方法。

Reviewer: ```suggestion
: Kubernetes RBAC 是一种根据企业内各个用户的角色来管理其对计算机或网络资源的访问权限的方法。
```

  在此上下文中，访问权限是单个用户执行特定任务（例如查看、创建或修改文件）的能力。
  在这种模式下，Kubernetes 使用 `rbac.authorization.k8s.io` API 组来驱动鉴权决策，
  允许你通过 Kubernetes API 动态配置权限策略。

`Node`
: 一种特殊用途的鉴权模式，根据 kubelet 计划运行的 Pod 向其授予权限。
  要了解有关 Node 鉴权模式的更多信息，请参阅 [Node 鉴权](/zh-cn/docs/reference/access-authn-authz/node/)。

`Webhook`
: Kubernetes 的 [Webhook 鉴权模式](/docs/reference/access-authn-authz/webhook/)用于鉴权，进行同步 HTTP 调用，

>   阻止请求直到远程 HTTP 服务响应查询。你可以编写自己的软件来处理调出，也可以使用生态系统中的解决方案。

Reviewer: ```suggestion
  阻塞请求直到远程 HTTP 服务响应查询。你可以编写自己的软件来处理这种向外调用，也可以使用生态系统中的解决方案。
```

<a id="warning-always-allow" />

{{< warning >}}
<!--
Enabling the `AlwaysAllow` mode bypasses authorization; do not use this on a cluster where
you do not trust **all** potential API clients, including the workloads that you run.

Authorization mechanisms typically return either a _deny_ or _no opinion_ result; see
[authorization verdicts](#determine-whether-a-request-is-allowed-or-denied) for more on this.
Activating the `AlwaysAllow` means that if all other authorizers return “no opinion”,
the request is allowed. For example, `--authorization-mode=AlwaysAllow,RBAC` has the
same effect as `--authorization-mode=AlwaysAllow` because Kubernetes RBAC does not
provide negative (deny) access rules.

You should not use the `AlwaysAllow` mode on a Kubernetes cluster where the API server
is reachable from the public internet.
-->
启用 `AlwaysAllow` 模式会绕过鉴权；请勿在你不信任**所有**潜在 API
客户端（包括你运行的工作负载）的集群上使用该模式。

鉴权机制通常返回“拒绝”或“无意见”的结果；
有关更多信息，请参阅[鉴权裁决](#determine-whether-a-request-is-allowed-or-denied)。

> 激活 `AlwaysAllow` 意味着如果所有其他鉴权者都返回 “no opinion”，则允许该请求。

Reviewer: ```suggestion
激活 `AlwaysAllow` 意味着如果所有其他鉴权者都返回“无意见”，则允许该请求。
```

例如，`--authorization-mode=AlwaysAllow,RBAC` 与 `--authorization-mode=AlwaysAllow`
具有相同的效果，因为 Kubernetes RBAC 不提供否定（拒绝）访问规则。

你不应在可从公共互联网访问 API 服务器的 Kubernetes 集群上使用 `AlwaysAllow` 模式。
{{< /warning >}}

<!--
### Authorization mode configuration {#choice-of-authz-config}

You can configure the Kubernetes API server's authorizer chain using either
[command line arguments](#using-flags-for-your-authorization-module) only or, as a beta feature,
using a [configuration file](#using-configuration-file-for-authorization).

You have to pick one of the two configuration approaches; setting both `--authorization-config`
path and configuring an authorization webhook using the `--authorization-mode` and
`--authorization-webhook-*` command line arguments is not allowed.
If you try this, the API server reports an error message during startup, then exits immediately.
-->
### 鉴权模式配置 {#choice-of-authz-config}

> 你可以仅使用[命令行参数](#using-flags-for-your-authorization-module)或作为 Beta
> 特性使用[配置文件](#using-configuration-file-for-authorization)来配置 Kubernetes API 服务器的鉴权链。

Reviewer: ```suggestion
你可以仅使用[命令行参数](#using-flags-for-your-authorization-module)，
或使用[配置文件](#using-configuration-file-for-authorization)来配置 Kubernetes API
服务器的鉴权链，后者目前是 Beta 特性。
```

你必须选择两种配置方法之一；不允许同时设置 `--authorization-config` 路径并使用
`--authorization-mode` 和 `--authorization-webhook-*` 命令行参数配置鉴权 Webhook。
如果你尝试这样做，API 服务器会在启动期间报告错误消息，然后立即退出。

<!--
### Command line authorization mode configuration {#using-flags-for-your-authorization-module}
-->
### 命令行鉴权模式配置  {#using-flags-for-your-authorization-module}

{{< feature-state state="stable" for_k8s_version="v1.8" >}}

<!--
You can use the following modes:

* `--authorization-mode=ABAC` (Attribute-based access control mode)
* `--authorization-mode=RBAC` (Role-based access control mode)
* `--authorization-mode=Node` (Node authorizer)
* `--authorization-mode=Webhook` (Webhook authorization mode)
* `--authorization-mode=AlwaysAllow` (always allows requests; carries [security risks](#warning-always-allow))
* `--authorization-mode=AlwaysDeny` (always denies requests)

You can choose more than one authorization mode; for example:
`--authorization-mode=Node,Webhook`
-->
你可以使用以下模式：

* `--authorization-mode=ABAC`（基于属性的访问控制模式）
* `--authorization-mode=RBAC`（基于角色的访问控制模式）

> * `--authorization-mode=Node`（节点鉴权者）

Reviewer: ```suggestion
* `--authorization-mode=Node`（节点鉴权组件）
```

-->
有关 API 服务器命令行参数的更多信息，请阅读
[`kube-apiserver` 参考](/zh-cn/docs/reference/command-line-tools-reference/kube-apiserver/)。

<!-- keep legacy hyperlinks working -->
<a id="configuring-the-api-server-using-an-authorization-config-file" />

<!--
### Configuring the API Server using an authorization config file {#using-configuration-file-for-authorization}
-->
### 使用鉴权配置文件配置 API 服务器 {#using-configuration-file-for-authorization}

{{< feature-state feature_gate_name="StructuredAuthorizationConfiguration" >}}

<!--
As a beta feature, Kubernetes lets you configure authorization chains that can include multiple
webhooks. The authorization items in that chain can have well-defined parameters that validate
requests in a particular order, offering you fine-grained control, such as explicit Deny on failures.

The configuration file approach even allows you to specify
[CEL](/docs/reference/using-api/cel/) rules to pre-filter requests before they are dispatched
to webhooks, helping you to prevent unnecessary invocations. The API server also automatically
reloads the authorizer chain when the configuration file is modified.
-->
作为一项 Beta 级别特性，Kubernetes 允许你配置可包含多个 Webhook 的鉴权链。

> 该链中的鉴权项可以具有明确定义的参数，这些参数可以按特定顺序验证请求，

Reviewer: ```suggestion
该链中的鉴权项可以具有明确定义的参数，这些参数可以按特定顺序检查请求，
```

`Webhook`
: Kubernetes 的 [Webhook 鉴权模式](/docs/reference/access-authn-authz/webhook/)用于鉴权，进行同步 HTTP 调用，
  阻塞请求直到远程 HTTP 服务响应查询。你可以编写自己的软件来处理这种向外调用，也可以使用生态系统中的解决方案。

<a id="warning-always-allow" />

{{< warning >}}
<!--
Enabling the `AlwaysAllow` mode bypasses authorization; do not use this on a cluster where
you do not trust **all** potential API clients, including the workloads that you run.

Authorization mechanisms typically return either a _deny_ or _no opinion_ result; see
[authorization verdicts](#determine-whether-a-request-is-allowed-or-denied) for more on this.
Activating the `AlwaysAllow` means that if all other authorizers return “no opinion”,
the request is allowed. For example, `--authorization-mode=AlwaysAllow,RBAC` has the
same effect as `--authorization-mode=AlwaysAllow` because Kubernetes RBAC does not
provide negative (deny) access rules.

You should not use the `AlwaysAllow` mode on a Kubernetes cluster where the API server
is reachable from the public internet.
-->
启用 `AlwaysAllow` 模式会绕过鉴权；请勿在你不信任**所有**潜在 API
客户端（包括你运行的工作负载）的集群上使用该模式。

鉴权机制通常返回“拒绝”或“无意见”的结果；
有关更多信息，请参阅[鉴权裁决](#determine-whether-a-request-is-allowed-or-denied)。

> 激活 `AlwaysAllow` 意味着如果所有其他鉴权者都返回“无意见”，则允许该请求。

Reviewer 2: ```suggestion
激活 `AlwaysAllow` 意味着如果所有其他鉴权器都返回“无意见”，则允许该请求。
```
“ authorizer” -> "鉴权器"
根据中文属于对照：https://kubernetes.io/zh-cn/docs/contribute/localization_zh/#glossary

下同。

Reviewer: “鉴权器”的翻译有点儿生硬，是不是意译为“鉴权组件”更合适？

Reviewer 2: 也可以。

to webhooks, helping you to prevent unnecessary invocations. The API server also automatically
reloads the authorizer chain when the configuration file is modified.
-->
作为一项 Beta 级别特性，Kubernetes 允许你配置可包含多个 Webhook 的鉴权链。
该链中的鉴权项可以具有明确定义的参数，这些参数可以按特定顺序检查请求，
从而为你提供细粒度的控制，例如在失败时明确拒绝。

配置文件方法甚至允许你指定 [CEL](/zh-cn/docs/reference/using-api/cel/)规则，
在将请求发送到 Webhook 之前对其进行预过滤，从而帮助你防止不必要的调用。
修改配置文件时，API 服务器还会自动重新加载鉴权链。

<!--
You specify the path to the authorization configuration using the
`--authorization-config` command line argument.

If you want to use command line arguments instead of a configuration file, that's also a valid and supported approach.
Some authorization capabilities (for example: multiple webhooks, webhook failure policy, and pre-filter rules)
are only available if you use an authorization configuration file.
-->
你可以使用 `--authorization-config` 命令行参数指定鉴权配置的路径。

如果你想使用命令行参数而不是配置文件，这也是一种有效且受支持的方法。
某些鉴权功能（例如：多个 Webhook、Webhook 失败策略和预过滤规则）仅在使用鉴权配置文件时可用。

<!--
#### Example configuration {#authz-config-example}
-->
#### 示例配置   {#authz-config-example}

{{< highlight yaml "linenos=false,hl_lines=2-4" >}}

> <!--

Reviewer 2: <img width="489" alt="image" src="https://github.com/kubernetes/website/assets/44460091/3ec084e6-fa0e-480a-81a0-1d5b8bd52c20">

      matchConditions:
      # expression represents the expression which will be evaluated by CEL. Must evaluate to bool.
      # CEL expressions have access to the contents of the SubjectAccessReview in v1 version.
      # If version specified by subjectAccessReviewVersion in the request variable is v1beta1,
      # the contents would be converted to the v1 version before evaluating the CEL expression.
      #
      # Documentation on CEL: https://kubernetes.io/docs/reference/using-api/cel/
      #
      # only send resource requests to the webhook
      - expression: has(request.resourceAttributes)
      # only intercept requests to kube-system
      - expression: request.resourceAttributes.namespace == 'kube-system'
      # don't intercept requests from kube-system service accounts
      - expression: !('system:serviceaccounts:kube-system' in request.user.groups)
  - type: Node
    name: node
  - type: RBAC
    name: rbac
  - type: Webhook
    name: in-cluster-authorizer
    webhook:
      authorizedTTL: 5m
      unauthorizedTTL: 30s
      timeout: 3s
      subjectAccessReviewVersion: v1
      failurePolicy: NoOpinion
      connectionInfo:
        type: InClusterConfig

> -->

Reviewer 2: <img width="426" alt="image" src="https://github.com/kubernetes/website/assets/44460091/fb4dcef2-1d0e-44d7-bc6f-19b55babc9db">

---
#
# 请勿按原样使用配置，这只是一个示例。
#
apiVersion: apiserver.config.k8s.io/v1beta1
kind: AuthorizationConfiguration
authorizers:
  - type: Webhook
    # 用于描述鉴权人的名称
    # 这明确用于监控机制的指标
    # 注意
    #   - 该字段的验证与今天的 K8s 标签的验证方式类似。
    # 必填，无默认值
    name: webhook
    webhook:

>       # 缓存来自 Webhook 鉴权者的“鉴权”响应的持续时间

Reviewer 2: ```suggestion
      # 缓存来自 Webhook 鉴权器的“鉴权”响应的持续时间
```

      # 与设置 `--authorization-webhook-cache-authorized-ttl` 标志相同
      # 默认值：5m0s
      authorizedTTL: 30s

>       # 缓存来自 Webhook 鉴权者的“未授权”响应的持续时间。

Reviewer 2: ```suggestion
      # 缓存来自 Webhook 鉴权器的“未授权”响应的持续时间。
```

      # 与设置 `--authorization-webhook-cache-unauthorized-ttl` 标志相同
      # 默认值：30s
      unauthorizedTTL: 30s
      # Webhook 请求超时
      # 允许的最大时间为 30 秒。
      # 必填，没有默认值。
      timeout: 3s
      # 要发送到 Webhook 并期望从 webhook 获得的 authorization.k8s.io SubjectAccessReview 的 API 版本。
      # 与设置 `--authorization-webhook-version` 标志相同
      # 必填，无默认值
      # 有效值：v1beta1、v1
      subjectAccessReviewVersion: v1
      # MatchConditionSubjectAccessReviewVersion 指定评估 CEL 表达式的 SubjectAccessReview 版本
      # 有效值：v1
      # 必填，无默认值
      matchConditionSubjectAccessReviewVersion: v1
      # 当 Webhook 请求无法完成或返回格式错误的响应或评估 matchConditions 时出现错误时，控制鉴权决定。
      # 有效值：

>       #   - NoOpinion：继续联系后续鉴权者，看其中是否有人允许该请求

Reviewer 2: ```suggestion
      #   - NoOpinion：继续联系后续鉴权器，看其中是否有人允许该请求
```

>       #   - Deny：拒绝请求而不咨询后续鉴权者

Reviewer 2: ```suggestion
      #   - Deny：拒绝请求而不咨询后续鉴权器
```

you would need to make sure the config file is in a format that Kubernetes {{< skew currentVersion >}}
can understand, before you upgrade the cluster. If you downgrade to {{< skew currentVersionAddMinor -1 >}},
you would need to set the configuration appropriately.
-->
使用配置文件配置鉴权链时，请确保所有控制平面节点具有相同的文件内容。升级/降级集群时，请记下 API 服务器配置。
例如，如果从 Kubernetes {{< skew currentVersionAddMinor -1 >}}
升级到 Kubernetes {{< skew currentVersion >}}，则需要确保配置文件的格式是
Kubernetes {{< skew currentVersion >}} 可以理解的，然后再升级集群。
如果降级到 {{< skew currentVersionAddMinor -1 >}}，则需要适当设置配置。

<!--
#### Authorization configuration and reloads

Kubernetes reloads the authorization configuration file when the API server observes a change
to the file, and also on a 60 second schedule if no change events were observed.
-->
#### 鉴权配置和重新加载

当 API 服务器观察到文件的更改时，Kubernetes 会重新加载鉴权配置文件，
如果没有观察到更改事件，则也会按照 60 秒的计划重新加载。

{{< note >}}
<!--
You must ensure that all non-webhook authorizer types remain unchanged in the file on reload.

A reload **must not** add or remove Node or RBAC authorizers (they can be reordered,
but cannot be added or removed).
-->

> 你必须确保重新加载时文件中所有非 Webhook 鉴权器类型保持不变。

Reviewer 2: ```suggestion
你必须确保重新加载时文件中所有非 Webhook 鉴权组件类型保持不变。
```
全文保持一致。

> 重新加载**不能**添加或删除节点或 RBAC 鉴权者（可以重新排序，但不能添加或删除）。

Reviewer 2: ```suggestion
重新加载**不能**添加或删除节点或 RBAC 鉴权器（可以重新排序，但不能添加或删除）。
```
